packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,366 @@
+   */
+  public static greaterThan(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.GREATER_THAN, target);
+  }
+
+  /**
+   * Convenience function to generate a LESS_THAN Filter
+   */
+  public static lessThan(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.LESS_THAN, target);
+  }
+
+  /**
+   * Convenience function to generate an EQUAL_TO Filter
+   */
+  public static equalTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate a NOT_EQUAL_TO Filter
+   */
+  public static notEqualTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.NOT_EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate an IS_PRESENT Filter
+   */
+  public static isPresent(match: string, target: boolean): Filter {
+    return new Filter(match, FilterOperator.IS_PRESENT, target);
+  }
+
+  /**
+   * The log event field to assess in the filter
+   */
+  readonly match: string;
+
+  /**
+   * The type of Filter to apply
+   */
+  readonly operator: FilterOperator;
+
+  /**
+   * The target field to compare the `match` against
+   */
+  readonly target: string[] | number | boolean;
+
+  private constructor(match: string, operator: FilterOperator, target: string[] | number | boolean) {
+    this.match = match;
+    this.operator = operator;
+    this.target = target;
+  }
+}
+
+/**
+ * The applicable Log Formats for the Insight Rule
+ */
+export enum LogFormat {
+  /**
+   * JSON format
+   */
+  JSON = 'JSON',
+
+  /**
+   * CLF format
+   */
+  CLF = 'CLF',
+}
+
+/**
+ * The options to aggregate the log events to evaluate contribution for the Insight Rule
+ */
+export enum AggregateOptions {
+  /**
+   * Aggregate contribution by summing a given field
+   */
+  SUM = 'Sum',
+
+  /**
+   * Aggregate contribution by the count of a contributor's appearance in events
+   */
+  COUNT = 'Count',
+}
+
+/**
+ * The properties for an Insight Rule
+ */
+export interface InsightRuleProps {
+  /**
+   * Option to Aggregate the Log Events for the Insight Rule to assess contribution
+   *
+   * @default - COUNT
+   */
+  readonly aggregateOn?: AggregateOptions,
+
+  /**
+   * Whether or not the Insight Rule is enabled
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The Filters to apply to the Insight Rule
+   *
+   * @default []
+   */
+  readonly filters?: Filter[],
+
+  /**
+   * The keys of the log event to identify a Contributor
+   */
+  readonly keys: string[],
+
+  /**
+   * The Log Format of the Insight Rule
+   *
+   * @default - JSON
+   */
+  readonly logFormat?: LogFormat,
+
+  /**
+   * The Log Groups to include in the report
+   */
+  readonly logGroupNames: string[],
+
+  /**
+   * The name of the Insight Rule
+   */
+  readonly insightRuleName: string,
```

Comment:
Going to leave this here for kaizen to review. Is this a spot we could/should use physicalName? https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_core.PhysicalName.html

Reply:
Not sure what you mean, although this does become the Physical Name on line 310. From the users perspective, they are supplying a name for the construct that does indeed get passed along to the physicalName of the construct via the `super` call on line 310. So if that's what you're worried about, then I think it's happening.